sympy/polys/matrices/ddm.py

```diff
@@ -92,6 +92,25 @@ def __init__(self, rowslist, shape, domain):
         if not (len(self) == m and all(len(row) == n for row in self)):
             raise DDMBadInputError("Inconsistent row-list/shape")
 
+    def getitem(self, i, j):
+        return self[i][j]
+
+    def extract_slice(self, slice1, slice2):
+        from .domainmatrix import getindex
+        m, n = self.shape
+        i, _m = getindex(slice1, m)
+        j, _n = getindex(slice2, n)
```

Comment:
The import shouldn't be here. I don't think it is needed anyway. Why can't we just use slice1 and slice2 directly?
```
ddm = [row[slice2] for row in self[slice1]]
```

Reply:
> Why can't we just use slice1 and slice2 directly?
> 
For now, we have implemented `DDM.extract_slice` independent of `DomainMatrix.__getitem__`. In `SDM.extract_slice`, it assumes `DomainMatrix.__getitem__` will call it and the `DomainMatrix.__getitem__` uses the `getindex` method before calling `SDM.extract_slice`.
Thus, we need to use `getindex` in `DDM.extract_slice`.

Kindly share your opinion.
Thanks.

Reply:
It should be possible to use the slices directly.

Reply:
> It should be possible to use the slices directly.

Only if we assume that `DomainMatrix.__getitem__` will call as it does for `SDM.extract_slice`.
Otherwise
```
In [1]: a=[1,2,3]

In [2]: a[slice(None, -3, None)]
Out[2]: []
```
This should have been `1` instead of an empty list.
Failing example: `dm.extract_slice(slice(1, 2), slice(-3))`

Kindly share your opinion.
Thanks.

Reply:
> Only if we assume that `DomainMatrix.__getitem__` will call as it does for `SDM.extract_slice`.

That's what is expected to happen. Both `SDM` and `DDM` will have identical `extract_slice` methods that will be used by `DomainMatrix`.

> This should have been `1` instead of an empty list.

No, it should be an empty list as shown. I'm not sure what you mean by that.

Reply:
I am extremely sorry, you were correct, it works without the `getindex` method.
Thanks a lot!